pkg/agent/proxy/proxier.go

```diff
@@ -355,40 +461,75 @@ func (p *proxier) installServices() {
 				klog.Errorf("Error when installing Service flows: %v", err)
 				continue
 			}
-			// Install OpenFlow entries for the ingress IPs of LoadBalancer Service.
-			// The LoadBalancer Service should be accessible from Pod, Node and
-			// external host.
-			var toDelete, toAdd []string
-			if needRemoval {
-				toDelete = pSvcInfo.LoadBalancerIPStrings()
-				toAdd = svcInfo.LoadBalancerIPStrings()
-			} else {
-				toDelete = deletedLoadBalancerIPs
-				toAdd = addedLoadBalancerIPs
-			}
-			for _, ingress := range toDelete {
-				if ingress != "" {
-					// It is safe to access pSvcInfo here. If this is a new Service,
-					// then toDelete will be an empty slice.
-					if err := p.uninstallLoadBalancerServiceFlows(net.ParseIP(ingress), uint16(pSvcInfo.Port()), pSvcInfo.OFProtocol); err != nil {
-						klog.Errorf("Error when removing LoadBalancer Service flows: %v", err)
+
+			if p.proxyAllEnabled {
+				// Install ClusterIP route on Node so that ClusterIP can be accessed on Node. Every time a new ClusterIP
+				// is created, the routing target IP block will be recalculated for expansion to be able to route the new
+				// created ClusterIP. Deleting a ClusterIP will not shrink the target routing IP block. The Service CIDR
+				// can be finally calculated after creating enough ClusterIPs.
+				if err := p.routeClient.AddClusterIPRoute(svcInfo.ClusterIP(), p.isIPv6); err != nil {
+					klog.ErrorS(err, "Failed to install ClusterIP route of Service", "Service", svcPortName)
+				}
+
+				// If externalTrafficPolicy of the Service is Local, Service NodePort or LoadBalancer should use the Service
+				// group whose Endpoints are local.
+				nGroupID := groupID
+				if svcInfo.NodeLocalExternal() {
+					nGroupID, _ = p.groupCounter.Get(svcPortName, true)
+				}
+
+				// If previous Service is nil or NodePort flows and configurations of previous Service have been removed,
+				// install NodePort flows and configurations for current Service.
+				if svcInfo.NodePort() > 0 && (pSvcInfo == nil || needRemoval) {
+					if err := p.installNodePortService(nGroupID, uint16(svcInfo.NodePort()), svcInfo.OFProtocol, uint16(svcInfo.StickyMaxAgeSeconds()), svcInfo.NodeLocalExternal()); err != nil {
+						klog.ErrorS(err, "Failed to install NodePort flows and configurations of Service", "Service", svcPortName)
 						continue
 					}
 				}
-			}
-			for _, ingress := range toAdd {
-				if ingress != "" {
-					if err := p.installLoadBalancerServiceFlows(groupID, net.ParseIP(ingress), uint16(svcInfo.Port()), svcInfo.OFProtocol, uint16(svcInfo.StickyMaxAgeSeconds())); err != nil {
-						klog.Errorf("Error when installing LoadBalancer Service flows: %v", err)
+
+				// Service LoadBalancer flows can be partially updated.
+				var toDelete, toAdd []string
+				if needRemoval {
```

Comment:
@tnqn I can change it back. Before the patch, when a Pod access an ingress IP of LoadBalancer, the the flows are used, at least in Linux. This part code is add by PR https://github.com/antrea-io/antrea/pull/943. I prefer to let LoadBalancer related flows be managed by proxyAll, so that the code looks more simpler. @lzhecheng, do you know how the LoadBalancer in Windows works?  Can Service LoadBalancer be accessed from outside the cluster?